opds.py

```diff
@@ -1,202 +1,97 @@
-import datetime
-import logging
-
-from lxml import builder, etree
 from nose.tools import set_trace
+import json
 
-class AtomFeed(object):
-
-    TIME_FORMAT = '%Y-%m-%dT%H:%M:%SZ%z'
-
-    ATOM_NS = 'http://www.w3.org/2005/Atom'
-    OPDS_NS = 'http://opds-spec.org/2010/catalog'
-    SCHEMA_NS = 'http://schema.org/'
-    
-    nsmap = {
-        None: ATOM_NS,
-        'opds' : OPDS_NS,
-        'schema' : SCHEMA_NS
-    }
-
-    default_typemap = {datetime: lambda e, v: _strftime(v)}
-    E = builder.ElementMaker(typemap=default_typemap, nsmap=nsmap)
-    SCHEMA = builder.ElementMaker(typemap=default_typemap, nsmap=nsmap, namespace=SCHEMA_NS)
-       
-    @classmethod
-    def _strftime(self, date):
-        """
-        Format a date the way Atom likes it (RFC3339?)
-        """
-        return date.strftime(self.TIME_FORMAT)
-
-
-    @classmethod
-    def add_link_to_feed(cls, feed, children=None, **kwargs):
-        link = cls.E.link(**kwargs)
-        feed.append(link)
-        if children:
-            for i in children:
-                link.append(i)
-
-
-    @classmethod
-    def add_link_to_entry(cls, entry, children=None, **kwargs):
-        link = cls.E.link(**kwargs)
-        entry.append(link)
-        if children:
-            for i in children:
-                link.append(i)
-
-
-    @classmethod
-    def author(cls, *args, **kwargs):
-        return cls.E.author(*args, **kwargs)
-
-
-    @classmethod
-    def category(cls, *args, **kwargs):
-        return cls.E.category(*args, **kwargs)
-
+class Annotator(object):
 
-    @classmethod
-    def entry(cls, *args, **kwargs):
-        return cls.E.entry(*args, **kwargs)
+    def annotate_feed(self, feed):
+        pass
 
+class OPDSCatalog(object):
```

Comment:
I think that's the right thing to do, but the recursion is another thing that confused me. Saying that this object might be an OPDS document on its own, or it might be contained within another OPDS document using the "catalogs" role, would help.